### ISSUE TITLE
Multiply activations of different shapes?

### ISSUE THREAD

**Issue by basedrhys** [deeplearning4j/deeplearning4j#8962]
#### Issue Description

Is there some way for a layer to multiply activations of different size?
I'm trying to load the EfficientNet family of models using DL4J, however, they use a "Squeeze and Excite" block:

![image](https://user-images.githubusercontent.com/21359576/82302700-0f971280-9a0e-11ea-8a61-a67bc8734f9b.png)

The `Scale` block multiplies 1x1xC activations with WxHxC activations, resulting in a WxHxC shape activations.
This is supported in the Keras Multiply layer:
```
> arr1 = np.random.rand(32, 112, 112)
> arr2 = np.random.rand(32, 1, 1)
> tf.keras.layers.Multiply()([arr1, arr2])

<tf.Tensor: shape=(32, 112, 112), dtype=float32, numpy=
array([[[0.08604141, 0.31940213, 0.08610599, ..., 0.16845039,
```
Note that this won't work if arr2 is (32, 2, 2).

Is there any way to get this working in DL4J? The ElementWiseVertex that is used for the Multiply op doesn't support this currently.

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version: 1.0.0-beta7
* Platform information (OS, etc): Windows
* CUDA version, if used
* NVIDIA driver version, if in use

#### Additional Information

Where applicable, please also provide:

Stacktrace: https://gist.github.com/43082099e51f91aa1b9f9deb367fbb10

**Comment by raver119** [comment]
This looks like a regular broadcasting, and definitely it should work. At least it definitely does in underlying libnd4j/ND4j.

cc @AlexDBlack

**Comment by AlexDBlack** [comment]
ElementWiseVertex forward pass itself should support that... but clearly `ElementWiseVertex.getOutputType(ElementWiseVertex.java:162)` itself does not.

If you were building the network from scratch (not via Keras import), there's an easy workaround in the form of SameDiffLambdaVertex (i.e., use that instead of ElementWiseVertex)... unfortunately that doesn't help here.

If you are building the net in Keras (instead of using someone else's model/code), adding a tf.broadcast_to before the scale op should provide a workaround (with a small memory use hit).

An alternative (bit of an ugly hack) if you can modify the Keras model would be to add an identity lambda after the sigmoid (i.e., just returns input unchanged in Keras), but then during DL4J Keras import, import those identity lambdas as ones that do the broadcast to the [minibatch,H,W,C] shape.
You can see lambdas being set for Keras import here:
https://github.com/eclipse/deeplearning4j/blob/master/deeplearning4j/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/e2e/KerasLambdaTest.java

Beyond that this use case isn't supported in DL4J yet, but it's something we should definitely add.
Sorry I don't have a better workaround than those.

**Comment by basedrhys** [comment]
Thanks for the quick replies, that sort of info is what I was after so I'll give the broadcasting a shot - thanks!
